homeassistant/components/motion_blinds/sensor.py

```diff
@@ -0,0 +1,178 @@
+        """Return the device class of this entity."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._blind.battery_level
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        attributes = {}
+        attributes[ATTR_BATTERY_VOLTAGE] = self._blind.battery_voltage
+        return attributes
+
+
+class MotionTDBUBatterySensor(MotionBatterySensor):
+    """
+    Representation of a Motion Battery Sensor for a Top Down Bottom Up blind.
+
+    Updates are done by the cover platform.
+    """
+
+    def __init__(self, coordinator, blind, motor):
+        """Initialize the Motion Battery Sensor."""
+        super().__init__(coordinator, blind)
+
+        self._motor = motor
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return f"{self._blind.mac}-{self._motor}-battery"
+
+    @property
+    def name(self):
+        """Return the name of the blind battery sensor."""
+        return f"{self._blind.blind_type}-{self._motor}-battery-{self._blind.mac}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._blind.battery_level is None:
+            return None
+        return self._blind.battery_level[self._motor[0]]
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        attributes = {}
+        if self._blind.battery_voltage is not None:
+            attributes[ATTR_BATTERY_VOLTAGE] = self._blind.battery_voltage[
+                self._motor[0]
+            ]
+        return attributes
+
+
+class MotionSignalStrengthSensor(CoordinatorEntity, Entity):
+    """Representation of a Motion Signal Strength Sensor."""
+
+    def __init__(self, coordinator, device, device_type):
+        """Initialize the Motion Signal Strength Sensor."""
+        super().__init__(coordinator)
+
+        self._device = device
+        self._device_type = device_type
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return f"{self._device.mac}-RSSI"
+
+    @property
+    def device_info(self):
+        """Return the device info of the blind."""
+        return {"identifiers": {(DOMAIN, self._device.mac)}}
+
+    @property
+    def name(self):
+        """Return the name of the blind signal strength sensor."""
+        if self._device_type == TYPE_GATEWAY:
+            return "Motion gateway signal strength"
+        return f"{self._device.blind_type} signal strength - {self._device.mac}"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return SIGNAL_STRENGTH_DECIBELS_MILLIWATT
+
+    @property
+    def device_class(self):
+        """Return the device class of this entity."""
+        return DEVICE_CLASS_SIGNAL_STRENGTH
```

Comment:
Usually we make these disabled by default as they tend to spam the state machine with updates.

https://developers.home-assistant.io/docs/core/entity/#advanced-properties

`entity_registry_enabled_default`

Reply:
Thanks, I prevented it from beeing added by default